packages/upload-api/src/access/claim.js

```diff
@@ -1,29 +1,31 @@
 import * as Server from '@ucanto/server'
 import * as Access from '@web3-storage/capabilities/access'
-import * as API from '../api.js'
+import * as API from '../types.js'
 import * as delegationsResponse from '../utils/delegations-response.js'
-import { collect } from 'streaming-iterables'
 
 /**
- * @typedef {object} Context
- * @property {object} models
- * @property {API.DelegationsStorage} models.delegations
- * @param {Context} ctx
+ * @param {API.AccessClaimContext} ctx
  */
 export const provide = (ctx) =>
   Server.provide(Access.claim, (input) => claim(input, ctx))
 
 /**
  * @param {API.Input<Access.claim>} input
- * @param {Context} ctx
+ * @param {API.AccessClaimContext} ctx
  * @returns {Promise<API.Result<API.AccessClaimSuccess, API.AccessClaimFailure>>}
  */
-export const claim = async ({ invocation }, { models: { delegations } }) => {
+export const claim = async (
+  { invocation },
+  { delegationsStorage: delegations }
+) => {
   const claimedAudience = invocation.capabilities[0].with
-  const claimed = await collect(delegations.find({ audience: claimedAudience }))
+  const claimedResult = await delegations.find({ audience: claimedAudience })
+  if (claimedResult.error){
+    return claimedResult
+  }
   return {
     ok: {
-      delegations: delegationsResponse.encode(claimed),
+      delegations: delegationsResponse.encode(claimedResult.ok),
```

Comment:
that's an interesting idea. definitely out of scope for this PR because of the client updates it would require, but I'm open to this - would probably save us a bit of money on bandwidth and compute.